pkg/scheduler/api/resource_info.go

```diff
@@ -480,27 +480,40 @@ func (r *Resource) SetScalar(name v1.ResourceName, quantity float64) {
 // e.g r resource is <cpu 2000.00, memory 4047845376.00, hugepages-2Mi 0.00, hugepages-1Gi 0.00>
 // rr resource is <cpu 3000.00, memory 1000.00>
 // return r resource is <cpu 2000.00, memory 1000.00, hugepages-2Mi 0.00, hugepages-1Gi 0.00>
-func (r *Resource) MinDimensionResource(rr *Resource) *Resource {
+// @param defaultValue "default value for resource dimension not defined in ScalarResources. Its value can only be one of 'Zero' and 'Infinity'"
+func (r *Resource) MinDimensionResource(rr *Resource, defaultValue DimensionDefaultValue) *Resource {
 	if rr.MilliCPU < r.MilliCPU {
 		r.MilliCPU = rr.MilliCPU
 	}
 	if rr.Memory < r.Memory {
 		r.Memory = rr.Memory
 	}
 
+	if r.ScalarResources == nil {
```

Comment:
no needs for the `== nil` comparation.

Reply:
Sorry, I do not understand what you mean.
I think it is necessary. Otherwise, `for name := range r.ScalarResources ` will cause panic.

Reply:
This is a kind-hearted redundancy. :)